pygsti/modelmembers/povms/__init__.py

```diff
@@ -498,14 +512,102 @@ def convert(povm, to_type, basis, ideal_povm=None, flatten_structure=False):
+                    
+                #Collect all ideal effects
+                base_dense_effects = []
+                for item in base_items:
+                    dense_effect = item[1].to_dense()
+                    base_dense_effects.append(dense_effect.reshape((1,len(dense_effect))))
+
+                dense_ideal_povm = _np.concatenate(base_dense_effects, axis=0)
+
+                #Collect all noisy effects
+                dense_effects = []
+                for effect in povm.values():
+                    dense_effect = effect.to_dense()
+                    dense_effects.append(dense_effect.reshape((1,len(dense_effect))))
+
+                dense_povm = _np.concatenate(dense_effects, axis=0)
+                
+                #It is often the case that there are more error generators than physical degrees of freedom in the POVM
+                #We define a function which finds linear comb. of errgens that span these degrees of freedom.
+                #This has been called "the trivial gauge", and this function is meant to avoid it
+                def calc_physical_subspace(dense_ideal_povm, epsilon = 1e-9):
+
+                    degrees_of_freedom = (dense_ideal_povm.shape[0] - 1) * dense_ideal_povm.shape[1]
+                    errgen = _LindbladErrorgen.from_error_generator(povm.state_space.dim, parameterization=to_type)
+
+                    if degrees_of_freedom > errgen.num_params:
+                        warnings.warn("POVM has more degrees of freedom than the available number of parameters, representation in this parameterization is not guaranteed")
+                    exp_errgen = _ExpErrorgenOp(errgen)
+                    
+                    num_errgens = errgen.num_params
+                    #TODO: Maybe we can use the num of params instead of number of matrix entries, as some of them are linearly dependent.
+                    #i.e E0 completely determines E1 if those are the only two povm elements (E0 + E1 = Identity)
+                    num_entries = dense_ideal_povm.shape[0]*dense_ideal_povm.shape[1]
+
+                    #Compute the jacobian with respect to the error generators. This will allow us to see which
+                    #error generators change the POVM entries
+                    J = _np.zeros((num_entries,num_errgens))
+                    
+                    for i in range(num_errgens):
+                        new_vec = _np.zeros(num_errgens)
+                        new_vec[i] = epsilon
+                        exp_errgen.from_vector(new_vec)
+                        vectorized_povm = _np.zeros(num_entries)
+                        perturbed_povm = (dense_ideal_povm @ exp_errgen.to_dense() - dense_ideal_povm)/epsilon 
+
+                        perturbed_povm_t = perturbed_povm.transpose()
+                        for j, column in enumerate(perturbed_povm_t):
+                            vectorized_povm[j*len(perturbed_povm_t[0]):(j+1)*len(perturbed_povm_t[0])] = column
```

Comment:
Minimum change: assign ``len(perturbed_povm_t[0])`` to a variable before the for-loop. Then use that variable in the loop for improved readability.

Preferred change: replace the for loop with ``vectorized_povm = perturbed_povm.flatten(order='F')``. Note that you don't need to predefine ``vectorized_povm = _np.zeros(num_entries)`` if you make this change.